lib/core/src/Cardano/Pool/DB/Sqlite.hs

```diff
@@ -212,15 +213,28 @@ newDBLayer trace fp = do
                     pure $ Just $ PoolRegistrationCertificate
                         { poolId, poolOwners, poolMargin, poolCost }
 
+        , putPoolRetirement = \point PoolRetirementCertificate
+            { poolId
+            , poolRetirementActiveBy
+            } -> do
+            -- Conversion functions treat 'NominalDiffTime' as seconds.
+            let timeSinceBlock0 = round poolRetirementActiveBy
+            insert_ $ PoolRetirement poolId point timeSinceBlock0
+
+        , readPoolRetirement = \poolId -> do
+            fmap (fromPoolRetirement . entityVal)
+                <$> selectFirst [ PoolRetirementPoolId ==. poolId ] []
```

Comment:
(For the part you now split off) There can't be multiple pool retirements, or something sneaky?